src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheFactory.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import co.rsk.panic.PanicProcessor;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PegoutSignedCacheFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger(PegoutSignedCacheFactory.class);
+  private static final PanicProcessor panicProcessor = new PanicProcessor();
+
+  private static PegoutSignedCache instance;
+
+  private PegoutSignedCacheFactory() {
+  }
+
+  /**
+   * Retrieves an instance of PegoutSignedCache with the specified Time To Live
+   * (TTL). If an instance does not already exist, it is lazily initialized.
+   * 
+   * @param ttl The duration for which the cache entries are considered valid.
+   * @return An instance of PegoutSignedCache.
+   * @throws IllegalArgumentException If the supplied TTL value is invalid.
+   */
+  public static PegoutSignedCache getInstance(final Duration ttl) throws IllegalArgumentException {
+    if (!isValidTtl(ttl)) {
+      String message = String.format(
+          "Invalid pegouts signed cache TTL value in minutes supplied: %d",
+          ttl != null ? ttl.toMinutes() : null);
```

Comment:
this check could be done in a variable too, `ttlInMinutes = ttl != null ? ttl.toMinutes() : null`

that way it can be reused when logging too :)

Reply:
With the new changes I don't think this applies anymore, wdyt?

Reply:
i would do it anyway, inside new `assertValidTtl` method

Reply:
Updated